apidef/oas/oas.go

```diff
@@ -112,24 +112,22 @@ func (s *OAS) getTykAuthentication() (authentication *Authentication) {
 }
 
 func (s *OAS) getTykTokenAuth(name string) (token *Token) {
-	if securitySchemes := s.getTykSecuritySchemes(); securitySchemes != nil {
-		securityScheme := securitySchemes[name]
-		if securityScheme == nil {
-			return
-		}
-
-		var ok bool
-		token, ok = securityScheme.(*Token)
-		if ok {
-			return
-		}
+	securityScheme := s.getTykSecurityScheme(name)
+	if securityScheme == nil {
+		return
+	}
 
+	mapSecurityScheme, ok := securityScheme.(map[string]interface{})
+	if ok {
 		token = &Token{}
-		inBytes, _ := json.Marshal(securityScheme)
+		inBytes, _ := json.Marshal(mapSecurityScheme)
```

Comment:
can we extract this code block to a separate method, since we'll be using it for all security mechanisms which can be configured with native OAS

Reply:
But we are casting to a different struct on each. How can we create a common function for all ?

Reply:
```suggestion
                func getStruct(mapSecurityScheme map[string]interface{}, target interface{}) {
                     inBytes, _ := json.Marshal(mapSecurityScheme)
		    _ = json.Unmarshal(inBytes, target)
                }
                mapSecurityScheme, ok := securityScheme.(map[string]interface{})
                if ok {
                   token = &Token{}
                   getStruct(mapSecurityScheme, token)
                    s.getTykSecuritySchemes()[name] = token
                }
```

Reply:
just the marhsalling and unmarshalling part would be in the function. or shall we thing of passing the securitySchemes as such? with security scheme name?

Reply:
I agree with the sentiment, but not the suggested implementation by jeffy, i think we can do better but let's address it later.